kitsune/wiki/views.py

```diff
@@ -471,106 +471,144 @@ def edit_document(request, document_slug, revision_id=None):
         url = reverse("wiki.translate", locale=request.LANGUAGE_CODE, args=[document_slug])
         return HttpResponseRedirect(url)
 
-    can_edit_needs_change = doc.allows(user, "edit_needs_change")
-    can_archive = doc.allows(user, "archive")
-
     # If this document has a parent, then the edit is handled by the
     # translate view. Pass it on.
     if doc.parent:
         return translate(request, doc.parent.slug, revision_id)
+
+    if not doc.allows(user, "create_revision"):
+        raise PermissionDenied
+
     if revision_id:
         rev = get_object_or_404(Revision, pk=revision_id, document=doc)
     else:
         rev = doc.current_revision or doc.revisions.order_by("-created", "-id")[0]
 
-    disclose_description = bool(request.GET.get("opendescription"))
-    doc_form = rev_form = None
-    if doc.allows(user, "create_revision"):
+    if request.method == "GET":
```

Comment:
Why is this needed?

Reply:
Usually we are checking against a post request that has some kind of a payload. GET requests should be the default and we just create the context and render the page. In my previous comment there was a suggestion to initialize the form for both if possible. 

For example you could do something like 
`rev_form = RevisionForm(request.POST or None, ....)

Reply:
Ok, after much toying with this - you can't avoid an explicit check for GET or POST, as there doesn't seem to be a way to designate a single rev_form definition without creating a problem.  Doing something like: `rev_form=RevisionForm(request.POST or None, instance=rev, initial={"based_on": rev.id, "comment": ""})` is problematic, since the initial values being defined will overwrite the POSTED data and break tests. You could assign the rev_form content based on a check for POST or GET, but that makes the code more difficult to understand AND still requires you to implicitly check for the request TYPE. Latest commit shows what I think is the best solution.